src/ingest-pipeline/airflow/dags/validate_upload.py

```diff
@@ -9,172 +9,245 @@
 from airflow.operators.python import PythonOperator
 from airflow.exceptions import AirflowException
 from airflow.providers.http.hooks.http import HttpHook
+from airflow.utils.email import send_email
 
 from hubmap_operators.common_operators import (
     CreateTmpDirOperator,
     CleanupTmpDirOperator,
 )
 
 from utils import (
-    get_tmp_dir_path, get_auth_tok,
+    encrypt_tok,
+    get_tmp_dir_path,
+    get_auth_tok,
     pythonop_get_dataset_state,
     HMDAG,
     get_queue_resource,
     get_preserve_scratch_resource,
-    get_threads_resource
-    )
+    get_threads_resource,
+)
+from catch_pipeline_errors import FailureCallback, FailureCallbackException
+
+sys.path.append(airflow_conf.as_dict()["connections"]["SRC_PATH"].strip("'").strip('"'))
+from submodules import (
+    ingest_validation_tools_upload,  # noqa E402
+    ingest_validation_tools_error_report,
+    ingest_validation_tests,
+)
 
-sys.path.append(airflow_conf.as_dict()['connections']['SRC_PATH']
-                .strip("'").strip('"'))
-from submodules import (ingest_validation_tools_upload,  # noqa E402
-                        ingest_validation_tools_error_report,
-                        ingest_validation_tests)
 sys.path.pop()
 
+
```

Comment:
Rather than importing the base FailureCallback class and having the ValidateUploadFailure class in here - consider moving the ValidateUploadFailure class into a separate file (maybe we have a directory where we define all of our failure classes). Reduces overall code in a singular file.

Reply:
Makes sense to me. I think a directory is probably the way to go and will create one.